pkg/distributor/http.go

```diff
@@ -44,22 +46,53 @@ func (d *Distributor) PushHandler(w http.ResponseWriter, r *http.Request) {
 func ParseRequest(r *http.Request) (*logproto.PushRequest, error) {
 	var req logproto.PushRequest
 
-	switch r.Header.Get(contentType) {
+	logger := util.WithContext(r.Context(), util.Logger)
+	body := lokiutil.NewSizeReader(r.Body)
+	contentType := r.Header.Get(contentType)
+
+	defer func() {
+		var (
+			entriesSize      int64
+			streamLabelsSize int64
+			totalEntries     int64
+		)
+
+		for _, s := range req.Streams {
+			streamLabelsSize += int64(len(s.Labels))
+			for _, e := range s.Entries {
+				totalEntries++
+				entriesSize += int64(len(e.Line))
+			}
+		}
+		level.Debug(logger).Log(
+			"msg", "push request parsed",
+			"path", r.URL.Path,
+			"content-type", contentType,
+			"body-size", humanize.Bytes(uint64(body.Size())),
+			"streams", len(req.Streams),
+			"entries", totalEntries,
+			"streamLabelsSize", humanize.Bytes(uint64(streamLabelsSize)),
```

Comment:
suggestion: keeping the log key names consistent?
either `contentType` and `bodySize` or
`stream-labels-size`, `entries-size` and `total-size`

currently its mix of both!